addons/project_todo/static/tests/delete_todo_stage_in_kanban_test.js

```diff
@@ -0,0 +1,79 @@
+/* @odoo-module */
+
+import { startServer } from "@bus/../tests/helpers/mock_python_environment";
+import { addModelNamesToFetch } from "@bus/../tests/helpers/model_definitions_helpers";
+
+import { start } from "@mail/../tests/helpers/test_utils";
+
+import {
+    editInput,
+    click,
+    getFixture,
+} from "@web/../tests/helpers/utils";
+import { setupViewRegistries } from "@web/../tests/views/helpers";
+
+addModelNamesToFetch(["project.project", "project.task", "project.task.type"]);
+
+let target;
+
+QUnit.module('kanban todo stage deletion', {
+    beforeEach: async function () {
+        const pyEnv = await startServer();
+        const projectId = pyEnv["project.project"].create([
+            { name: "Project One" },
+        ]);
+        const userId = pyEnv["res.users"].create([
+            { name: "User One", login: 'one', password: 'one' },
+        ]);
+        const stageId = pyEnv["project.task.type"].create([
+            { name: "Personal stage 1", user_id: userId },
+        ]);
+        pyEnv["project.task"].create([
+            { name: 'task one', project_id: projectId, personal_stage_type_id: stageId, user_ids: [userId] },
+        ]);
+        target = getFixture();
+        setupViewRegistries();
+    }
+}, function () {
+    QUnit.test("Check whether the stage are correctly deleted ", async function (assert) {
+
+        const views = {
+           "project.task,false,kanban":
+                `<kanban
+                default_group_by="personal_stage_type_id"
+                class="o_kanban_small_column"
+                js_class="project_task_kanban">
+                    <field name="project_id"/>
+                    <templates>
+                        <t t-name="kanban-box">
+                            <div>
+                                <field name="name"/>
+                                <field name="user_ids" invisible="1" widget="many2many_avatar_user"/>
+                            </div>
+                        </t>
+                    </templates>
+                </kanban>`,
+        };
+        const { openView } = await start({ serverData: { views } });
+        await openView({
+            res_model: "project.task",
+            views: [[false, "kanban"]],
+        });
+        // Adds an empty stage (its easier to create an empty stage here than to create a mock server to simulate the call to _read_group_personal_stage_type_ids)
+        await click(target, "button.o_kanban_add_column");
+        await editInput(target, "input.form-control", "Empty stage");
+        await click(target, "button.o_kanban_add");
```

Comment:
FYI, there are some helpers for the kanban view. 🙂 

```suggestion
        await createColumn(target);
        await editColumnName(target, "Empty stage");
        await validateColumn(target);
```

to import like this:

```js
import {
    createColumn,
    editColumnName,
    toggleColumnActions,
    validateColumn,
} from "@web/../tests/views/kanban/helpers";
```